doc/man1/roar-config.1
.\" roar-config.1:

.TH "roar-config" "1" "January 2010" "RoarAudio" "System User's Manual: roar-config"

.SH NAME

roar-config \- get information of installed RoarAudio librarys

.SH SYNOPSIS

roar-config [\-\-version] [\-\-libs] [\-\-cflags] [lib]

.SH DESCRIPTION

This small tool prints information about installed versions and needed flags to work
with the RoarAudio librarys.

.SH "OPTIONS"

.TP
\fB--version\fR
Show version of library

.TP
\fB--libs\fR
Show linker flags (\-lxxx) needed to link library

.TP
\fB--cflags\fR
Show compiler flags needed to link library

.SH "SEE ALSO"
\fBroartips\fR(7),
\fBlibroar\fR(7),
\fBRoarAudio\fR(7).

.SH "HISTORY"

For history information see \fBRoarAudio\fR(7).

.\" ll


ChangeLog
v. 0.3beta7 - ?
	* Added support for audio profiles
	* Updated libroarrsound
	* Added RPlay protocol to roard

v. 0.3beta6 - Thu Jun 24 2010 20:47 CEST
	Prereleases: 0: Fri Jun 11 2010 10:47 CEST;
	             1: Sat Jun 12 2010 11:07 CEST;
	             2: Wed Jun 16 2010 12:52 CEST
	* Added something like select() for VIOs
	* Fixed a lot warnings
	* disabled cmd codecfilter by default
	* Fixed error with roarinterconnect on non-OSS systems
	* Fixed some problems with minimal builds
	* Some Copyright fixes
	* much better VIO support for connections
	* much better OSS emulation
	* Updated win32 port
	* Updated RSound support
	* fixed pos calc in case of PCM->PCM converting (pr1)
	* fixed libroarpulse (pr2)

v. 0.3beta5 - Sun May 02 2010 12:41 CEST
	* added tool roardtmf
	* Support to store server address in X11's root window propertys
	* some small fixes
	* wrote a lot more libroarpulse code, some async API clients work now
	* added compbins for esddsp, artsdsp, audiooss, padsp
	* splited libroarpulse and libroarpulse-simple
	* upgraded some code to use VIO
	* more updates for Debian
	* fixed init script for Debian and Debian like systems
	* fixed some mixer stuff, a lot things work better now
	* added support for notify beeps
	* updated drivers
	* updated ROAR_{ERR,WARN,INFO,DBG}()
	* added syslog support
	* updated test suit
	* done a lot work on libroaross
	* fixed segfault in ALSA plugin
	* Added support for OSS to roarinterconnect(1)
	* Added server side RSound Emulation
	* Added libroarrsound, client side RSound Emulation
	* Added ALSA driver
	* Added RSound driver
	* Added PortAudio driver

v. 0.3beta4 - Sat Feb 06 2010 26:02 CET
	* Applyed patches for Debian
	* updated roarify to load libs of correct version
	* updated list of supported versions (complibs)
	* added header with common units
	* added support for PulseAudio Simple protocol
	* added support to libroaross to emulate DMX4Linux (write only)
	* added DSTR driver to roard
	* updated librarys' soname
	* do not link OpenSSL by default (licence incompatible?)
	* support for dynamic lib loading, plugin interface
	* support to for registering DSTR types
	* moved WinSock things out of main vio.[ch]
	* support auto-tool --prefix*=*

v. 0.3beta3 - Fri Jan 22 2010 22:17 CET
	* added some aRts/NAS/YIFF/PulseAudio tools to compbins
	* added support for comp libs to roar-config
	* added support for uni-directional streams in roarinterconnect
	* write much better support for config, including codec config
	* reduced the TCP overhead produced by speex
	* added support for speex preprocessor
	* get names of complibs from table, do not guess them
	* a lot small fixes
	* moved most of the mixer stuff out of roard into libroardsp
	* added flags immutable and enhance
	* test for signal.h
	* Corrected (DSTR) HTTP Protocol support so servers without headerflush
	  should now work, too.
	* added protocol support for icy
	* Added version numbers to lib names
	* updated manpages
	* wrote OSS emulation
	* fixed test scripts

v. 0.3beta2 - Mon Oct 05 2009 19:48 CEST
	* Added prethru flag
	* Added support to read info about client's network node
	* Ported libroaresd, libroarartsc
	* Fixed some problems with EsounD emulation
	* added support for compatibility binarys
	* added poll() support to libroarsndio
	* reject streams of unsupported codec

v. 0.3beta1 - Sat Sep 26 2009 25:50 CEST
	* incremented nummber of streams per client fron 4 to 8
	* RIFF WAVE codecfilter uses virtual streams
	* another set of changed names of parameters in roarctl
	* Interface to read byteorder of a client from roard
	* implemented a 'state of stream' var
	* cleanup, small fixes
	* added VIO interface to transcoder interface
	* ask codecfilter of parent in case of new child stream
	* added 'subsystem' complex: multi subsystem data on single streams
	* added basic /Radio Data and Trasmitter Control System/.
	* better support for stereo upmixing
	* make audio group a optional thing
	* better support for win32
	* wrote WMM driver
	* better multi protocol support
	* Support for server side EsounD emulation on win32

v. 0.3beta0 - Sun Sep 06 2009 22:42 CEST
	* added new stream directions for MIDI and light control
	* cleanup of client stream struct
	* added basic light control subsystem
	* added DMX512 driver
	* wrote basic MIDI subsystem
	* added flag 'mute'
	* added flag 'mmap'
	* added lib libroarmidi and libroarlight
	* added OpenSLP support
	* rewote fillmixbuer2 code
	* added a roar_conv2() with a more clear API
	* added support to write pidfile
	* added support for encoding 32 bit Vorbis streams
	* added transcoding features in libroardsp
	* added tool 'roarphone'
	* added tool 'roarinterconnect'
	* added simple raw data subsystem
	* added working support for THRU streams
	* changed names of parameters in roarctl
	* added lib libroareio - Extended IO for RoarAudio
	* Update order of default drivers, now it is: oss, ao, esd, null
	* cleanup of old sysio driver interface
	* added server side EsounD emulation
	* added multi homed support to server
	* fixed a long outstanding bug with pa_path_get_filename()

v. 0.2 - Thu May 21 2009 21:00
	* do a some more testings for functions not existing on win32
	* init winsock
	* done a lot fixes in order to get libroar and some clients work on win32
	* done a lot cleanup on configure script

v. 0.2rc2 - Sun May 17 2009 25:44
	* corrected check for GNU peercred
	* added basic logging support to configure
	* fixed plugin configures
	* added and fixed semi-install target on plugins
	* call plugin configures from main configure
	* include plugins in Makefile known to work from configure
	* fixed Makefiles of plugins: added -I../../include, -L../../lib
	* Make use of pkg-config in plugin configures
	* corrected possition of -ls on linker commands in most Makefiles
	* added more explicide linker commands so most things now build
	  with classic BSD make.
	* devided into components: libs, roard, clients, doc

roarclients/roar-config.c
//roar-config.c:

/*
 *      Copyright (C) Philipp 'ph3-der-loewe' Schafft - 2008-2010
 *
 *  This file is part of roarclients a part of RoarAudio,
 *  a cross-platform sound system for both, home and professional use.
 *  See README for details.
 *
 *  This file is free software; you can redistribute it and/or modify
 *  it under the terms of the GNU General Public License version 3
 *  as published by the Free Software Foundation.
 *
 *  RoarAudio is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *  GNU General Public License for more details.
 *
 *  You should have received a copy of the GNU General Public License
 *  along with this software; see the file COPYING.  If not, write to
 *  the Free Software Foundation, 51 Franklin Street, Fifth Floor,
 *  Boston, MA 02110-1301, USA.
 *
 */

#include <roaraudio.h>

struct {
 char * name;
 char * cflags;
 char * libs;
} flags[] = {
 // native/own libs:
 {"roar",      ROAR_CFLAGS, ROAR_LIBS        }, // NOTE: libroar *MUST* be the first entry
 {"roardsp",   ROAR_CFLAGS, ROAR_LIBS_DSP    },
 {"roarmidi",  ROAR_CFLAGS, ROAR_LIBS_MIDI   },
 {"roarlight", ROAR_CFLAGS, ROAR_LIBS_LIGHT  },
 {"roareio",   ROAR_CFLAGS, ROAR_LIBS_EIO    },
 // comp libs:
 {"roaresd",   ROAR_CFLAGS, ROAR_LIBS_C_ESD  },
 {"esd",       ROAR_CFLAGS, ROAR_LIBS_C_ESD  },
 {"roarartsc", ROAR_CFLAGS, ROAR_LIBS_C_ARTSC},
 {"artsc",     ROAR_CFLAGS, ROAR_LIBS_C_ARTSC},
 {"roarpulse", ROAR_CFLAGS, ROAR_LIBS_C_PULSE},
 {"pulse",     ROAR_CFLAGS, ROAR_LIBS_C_PULSE},
 {"roarpulse-simple", ROAR_CFLAGS, ROAR_LIBS_C_PULSE_SIMPLE},
 {"pulse-simple",     ROAR_CFLAGS, ROAR_LIBS_C_PULSE_SIMPLE},
 {"roarsndio", ROAR_CFLAGS, ROAR_LIBS_C_SNDIO},
 {"sndio",     ROAR_CFLAGS, ROAR_LIBS_C_SNDIO},
 {"roaryiff",  ROAR_CFLAGS, ROAR_LIBS_C_YIFF },
 {"Y2",        ROAR_CFLAGS, ROAR_LIBS_C_YIFF },
 {NULL, NULL, NULL}
}, * flags_ptr = NULL;

void usage (void) {
 printf("Usage: roar-config [--version] [--libs] [--cflags] [lib]\n");

 printf("\nOptions:\n\n");

 printf(
        "  --version          - Show version of library\n"
        "  --libs             - Show linker flags (-lxxx) needed to link library\n"
        "  --cflags           - Show compiler flags needed to link library\n"
        "  --output-pc        - Output PC format\n"
        "  --output-normal    - Output PC format\n"
       );

}

int main (int argc, char * argv[]) {
 enum { NORMAL, PC } mode = NORMAL;
 int i, h;
 int cflags = 0;
 int libs   = 0;
 char buf[1024] = {0};

 if ( argc == 1 ) {
  usage();
  return 0;
 }

 for (i = 1; i < argc; i++) {
  if ( !strcmp(argv[i], "--version") ) {
   printf("unknown\n");
  } else if ( !strcmp(argv[i], "--help") || !strcmp(argv[i], "-h") ) {
   usage();
   return 0;
  } else if ( !strcmp(argv[i], "--libs") ) {
   libs   = 1;
  } else if ( !strcmp(argv[i], "--cflags") ) {
   cflags = 1;
  } else if ( !strcmp(argv[i], "--output-normal") ) {
   mode = NORMAL;
  } else if ( !strcmp(argv[i], "--output-pc") ) {
   mode = PC;
  } else if ( flags_ptr == NULL ) {
   if ( !strncmp(argv[i], "lib", 3) )
    argv[i] += 3;

   for (h = 0; flags[h].name != NULL; h++) {
    if ( !strcasecmp(argv[i], flags[h].name) )
     flags_ptr = &(flags[h]);
   }

   if ( flags_ptr == NULL ) {
    fprintf(stderr, "Unknown lib: %s", argv[i]);
    return 2;
   }
  } else {
   fprintf(stderr, "Unknown option: %s\n", argv[i]);
   usage();
   return 1;
  }
 }

 if ( flags_ptr == NULL )
  flags_ptr = &(flags[0]);

 switch (mode) {
  case NORMAL:
    if ( cflags )
     strcat(buf, flags_ptr->cflags);

    if ( libs )
     strcat(buf, flags_ptr->libs);

    puts(buf);
   break;
  case PC:
    printf(
           "prefix=%s\n"
           "exec_prefix=${prefix}\n"
           "libdir=%s\n"
           "includedir=%s\n",
           PREFIX, PREFIX_LIB, PREFIX_INC
          );
    printf("\n");
    printf(
           "Name: lib%s\n"
           "Description: lib%s is part of RoarAudio Sound System\n"
           "Version: %s\n"
//           "Requires: libroar\n"
           "Conflicts:\n"
           "Libs: -L${libdir} %s\n"
           "Cflags: -I${includedir} %s\n",
           flags_ptr->name,
           flags_ptr->name,
           COMMON_VERSION,
           flags_ptr->libs,
           flags_ptr->cflags
          );
   break;
 }

 return 0;
}

//ll
